src/Compilers/Core/MSBuildTask/LocateEditorConfigFiles.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    /// <summary>
+    /// Locates the applicable set of .editorconfig files based on given inputs. Walks through
+    /// parent directories locating the ones that exist.
+    /// </summary>
+    public sealed class LocateEditorConfigFiles : Task
+    {
+        /// <summary>
+        /// The set of source and other files that we should find the applicable .editorconfig files to.
+        /// </summary>
+        [Required]
+        public ITaskItem[] InputFiles { get; set; }
```

Comment:
:memo: It is possible that this approach could fail to locate **.editorconfig** files - specifically in cases where the path mapping results in "overlaid" directory structures.